### ISSUE TITLE
Stop suggesting macro export as arbitraty ordering

### ISSUE THREAD

**Issue by eirnym** [rust-lang/rust-clippy#15572]
### Summary

There's 2 ways of exporting macro: using `#[macro_export]` annotation and `pub(scope) use macro_name`. The second way allows to export inside crate only and **must** go after macro definition, otherwise program won't compile.

### Lint Name

clippy::arbitrary-source-item-ordering

### Reproducer

I tried this code:

```rust
macro_rules! my_macro { 
 () => {}
}

pub(crate) use my_macro;
```

I saw this happen:

```
src/macro.rs:5:16
    |
102 | pub(crate) use my_macro;
    |                ^^^^^^^^^^^^^^
    |
note: should be placed before `my_macro`
   --> src/macro.rs:1:14
    |
12  | macro_rules! my_macro {
    |              ^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#arbitrary_source_item_ordering
    = note: `-W clippy::arbitrary-source-item-ordering` implied by `-W clippy::restriction`
    = help: to override `-W clippy::restriction` add `#[allow(clippy::arbitrary_source_item_ordering)]`

```

I expected to see this happen:
noting

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: aarch64-apple-darwin
release: 1.88.0
LLVM version: 20.1.5
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
